packages/client/hmi-client/src/components/models/tera-model-diagram.vue

```diff
@@ -0,0 +1,624 @@
+const equationLatexNew = ref<string>('');
+const isMathMLValid = ref<boolean>(true);
+
+const splitterContainer = ref<HTMLElement | null>(null);
+const layout = ref<'horizontal' | 'vertical' | undefined>('horizontal');
+
+const switchWidthPercent = ref<number>(50); // switch model layout when the size of the model window is < 50%
+
+const equationPanelSize = ref<number>(50);
+const equationPanelMinSize = ref<number>(0);
+const equationPanelMaxSize = ref<number>(100);
+
+const mathPanelSize = ref<number>(50);
+const mathPanelMinSize = ref<number>(0);
+const mathPanelMaxSize = ref<number>(100);
+
+const graphElement = ref<HTMLDivElement | null>(null);
+let renderer: PetrinetRenderer | null = null;
+let eventX = 0;
+let eventY = 0;
+
+const updateLayout = () => {
+	if (splitterContainer.value) {
+		layout.value =
+			(splitterContainer.value.offsetWidth / window.innerWidth) * 100 < switchWidthPercent.value ||
+			window.innerWidth < 800
+				? 'vertical'
+				: 'horizontal';
+	}
+};
+
+const handleResize = () => {
+	updateLayout();
+};
+
+onMounted(() => {
+	window.addEventListener('resize', handleResize);
+	handleResize();
+	// new model
+	if (props.assetId === '') {
+		isEditingEQ.value = true;
+		isMathMLValid.value = false;
+	}
+});
+
+onUnmounted(() => {
+	window.removeEventListener('resize', handleResize);
+});
+
+const mathEditorSelected = computed(() => {
+	if (!isMathMLValid.value) {
+		return 'math-editor-error';
+	}
+	if (isEditingEQ.value) {
+		return 'math-editor-selected';
+	}
+	return '';
+});
+
+const onVariableSelected = (variable: string) => {
+	if (variable) {
+		if (variable === selectedVariable.value) {
+			selectedVariable.value = '';
+			equationLatex.value = equationLatexOriginal.value;
+		} else {
+			selectedVariable.value = variable;
+			equationLatex.value = equationLatexOriginal.value.replaceAll(
+				selectedVariable.value,
+				String.raw`{\color{red}${variable}}`
+			);
+		}
+		renderer?.toggoleNodeSelectionByLabel(variable);
+	} else {
+		equationLatex.value = equationLatexOriginal.value;
+	}
+};
+
+const setNewLatexFormula = (formulaString: string) => {
+	equationLatexNew.value = formulaString;
+};
+
+const updateLatexFormula = (formulaString: string) => {
+	equationLatex.value = formulaString;
+	equationLatexOriginal.value = formulaString;
+};
+
+const cancelEditng = () => {
+	isEditingEQ.value = false;
+	isMathMLValid.value = true;
+	updateLatexFormula(equationLatexOriginal.value);
+};
+
+const fetchRelatedTerariumArtifacts = async () => {
+	if (model.value) {
+		const results = await getRelatedArtifacts(props.assetId, ProvenanceType.ModelRevision);
+		relatedTerariumArtifacts.value = results;
+	} else {
+		relatedTerariumArtifacts.value = [];
+	}
+};
+// Whenever selectedModelId changes, fetch model with that ID
+watch(
+	() => [props.assetId],
+	async () => {
+		updateLatexFormula('');
+		if (props.assetId !== '') {
+			const result = await getModel(props.assetId);
```

Comment:
Okay so it looks like the model should be passed into this instead of the id. The model is needed in the parent because we need to display the model name, description etc which is missing